src/hooks/useFetch.js

```diff
@@ -1,25 +1,37 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useRef } from 'react';
 
 const useFetch = (url, lat, lng) => {
-  const [data, setData] = useState([]);
+  const cache = useRef({});
+  const [locations, setLocations] = useState([]);
+  const [centerMap, setCenterMap] = useState({});
   const [status, setStatus] = useState('idle');
   const [error, setError] = useState(null);
+  const [zoom, setZoom] = useState(12);
 
   useEffect(() => {
     if (!url) return;
 
     const getWikiEntries = async (lat, lng) => {
       setStatus('loading');
       try {
-        const res = await fetch(url);
+        if (cache.current[url]) {
+          const data = cache.current[url];
+          setLocations(data);
+          setStatus('fetched');
+        } else {
+          const res = await fetch(url);
 
-        if (!res.ok) {
-          throw new Error(res.status);
-        }
+          if (!res.ok) {
+            throw new Error(res.status);
+          }
 
-        const data = await res.json();
-        setData({ ...data, centerCoords: { lat, lng }, zoom: 16 });
-        setStatus('success');
+          const response = await res.json();
```

Comment:
This is no longer a response, this is parsed data😄. Also, I'm curious why have you changed the lines containing object spreading?

Reply:
@katia-utochkina : I'm sorry this is my error. we don't have more of this file, we're using `useLocations` now.